components/org.wso2.carbon.identity.mgt.store.connector.jdbc/src/main/java/org/wso2/carbon/identity/mgt/store/connector/jdbc/connector/JDBCIdentityStoreConnector.java

```diff
@@ -527,20 +526,33 @@ public IdentityStoreConnectorConfig getIdentityStoreConfig() {
 
         if (databaseType != null && (databaseType.equalsIgnoreCase("MySQL") ||
                 databaseType.equalsIgnoreCase("H2"))) {
-
-            sqlQuerryForUserAttributes = new MySQLFamilySQLQueryFactory()
-                    .getQuerryForUserIdFromMultipleAttributes(attributes, offset, length);
-
+            if (attributes.size() > 0) {
+                sqlQuerryForUserAttributes = new MySQLFamilySQLQueryFactory()
+                        .getQuerryForUserIdFromMultipleAttributes(attributes, offset, length);
+            } else {
+                throw new IdentityStoreConnectorException("There are no attributes to find users.");
+            }
             if (log.isDebugEnabled()) {
                 log.debug("{} sql queries loaded for database type: {}.", sqlQueries.size(), databaseType);
             }
         } else {
-            throw new StoreException("Invalid or unsupported database type specified in the configuration.");
+            throw new IdentityStoreConnectorException("Invalid or unsupported database type specified in the" +
+                    " configuration.");
```

Comment:
Error messages need more context than this. Please mention the `databaseType` that the user has set.